File: domain-modules/bulo-flows-module/bulo-flows-data-module/src/main/java/com/maukaim/bulo/flows/data/models/flow/FlowStage.java
package com.maukaim.bulo.flows.data.models.flow;

import com.maukaim.bulo.commons.models.ContextStageId;

import java.util.Objects;
import java.util.Set;

public class FlowStage {
    private final ContextStageId contextStageId;
    private final Set<IoDependency> ioDependencies;

    public FlowStage(ContextStageId contextStageId, Set<IoDependency> ioDependencies) {
        this.contextStageId = contextStageId;
        this.ioDependencies = ioDependencies;
    }

    public ContextStageId getFlowStageId() {
        return contextStageId;
    }

    public Set<IoDependency> getIoDependencies() {
        return ioDependencies;
    }

    @Override
    public String toString() {
        return "FlowStage{" +
                "flowStageId=" + contextStageId +
                ", ioDependencies=" + ioDependencies +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        FlowStage flowStage = (FlowStage) o;
        return contextStageId.equals(flowStage.contextStageId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(contextStageId);
    }
}

File: front-end/lib/domains/flows/model/flow_stage.dart
import 'flow_stage_id.dart';
import 'io_dependency.dart';

class FlowStage {
  final FlowStageId flowStageId;
  final List<IoDependency> ioDependencies;

  FlowStage({required this.flowStageId, required this.ioDependencies});

  factory FlowStage.fromJson(Map<String, dynamic> json) {
    return FlowStage(
      flowStageId: FlowStageId.fromJson(json['flowStageId']),
      ioDependencies: [
        for (var x in json['ioDependencies']) IoDependency.fromJson(x)
      ],
    );
  }
}
